libcnb-test/src/test_runner.rs

```diff
@@ -101,6 +101,15 @@ impl TestRunner {
     ) {
         let config = config.borrow();
 
+        let mut test_context = TestContext {
```

Comment:
[Question, no action] Does this also close #588 ? My read is that yes, it does. 

I know you talked about wanting to wait on any large refactors in that PR before this was through (as this one is large as is).

Reply:
yes and no.  this would close #586. i think refactoring how the test context is tracked/cleaned up (e.g.; attaching docker cleanup logic to an `ImageName` struct) should be addressed in #588 where that discussion is happening.